src/node_sqlite.cc

```diff
@@ -1175,6 +1175,37 @@ void DatabaseSync::CustomFunction(const FunctionCallbackInfo<Value>& args) {
   CHECK_ERROR_OR_THROW(env->isolate(), db, r, SQLITE_OK, void());
 }
 
+void DatabaseSync::Location(const FunctionCallbackInfo<Value>& args) {
+  DatabaseSync* db;
+  ASSIGN_OR_RETURN_UNWRAP(&db, args.This());
+  Environment* env = Environment::GetCurrent(args);
+  THROW_AND_RETURN_ON_BAD_STATE(env, !db->IsOpen(), "database is not open");
+
+  std::string_view db_name = "main";
+  if (!args[0]->IsUndefined()) {
+    if (!args[0]->IsString()) {
+      THROW_ERR_INVALID_ARG_TYPE(env->isolate(),
+                                 "The \"dbName\" argument must be a string.");
+      return;
+    }
+
+    Utf8Value db_name_utf8(env->isolate(), args[0].As<String>());
+    db_name = *db_name_utf8 ? *db_name_utf8 : "";
+  }
```

Comment:
This isn't going to work as the assigned string_view will become invalid once the block exits, leading to sadness. The easiest thing to do here is to just revert `db_name` back to being a `std::string` as you had it previously. Let's do that and we can explore other ways to optimize later.

Reply:
Will do.

(I learned a lot o things anyways <3)